src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +29,79 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        // Check if no filters are given
+        if (args.trim().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(
+                        args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_PRIORITY, PREFIX_REMARK, PREFIX_TAG);
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        // Check for filtering by invalid prefixes
+        if (hasPrefixes(argMultimap, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_REMARK, PREFIX_TAG)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        List<String> names = argMultimap.getAllValues(PREFIX_NAME);
+        List<String> addresses = argMultimap.getAllValues(PREFIX_ADDRESS);
+        List<String> priorities = argMultimap.getAllValues(PREFIX_PRIORITY);
+
+        if (!areValidKeywords(names, Name.VALIDATION_REGEX)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE);
+        }
+
+        if (!areValidKeywords(addresses, Address.VALIDATION_REGEX)) {
+            throw new ParseException(Address.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE);
+        }
+
+        if (!areValidPriorities(priorities)) {
+            throw new ParseException(Priority.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE);
+        }
```

Comment:
Actually, I thought about this while designing too. Personally, I just felt that using the find command with invalid inputs, like those that do not match the specifications for a name or a prefix followed by no value, should throw an error rather than return an empty list or the entire list, because it gives the user feedback on how to properly pass filtering keywords for each prefix when using the find command and may also give more meaning to the find command itself - For instance, without throwing an error, if we permit "find a/" (blank value) , the program will simply list every contact, which is what the list command is meant for and find should be used to narrow down this list in some manner.

Reply:
I agree. If we didn't validate the keywords, `find n/ a/` would display all persons, which could confuse the user with `list`.